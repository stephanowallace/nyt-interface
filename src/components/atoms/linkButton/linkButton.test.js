import React from 'react';
import { shallow } from 'enzyme';
import LinkButton from '.';

describe('Link button', () => {
  it('renders default state', () => {
    const component = shallow(<LinkButton>test</LinkButton>);
    expect(component).toMatchSnapshot();
  });

  it('calls on click', () => {
    const onClick = jest.fn();
    const component = shallow(<LinkButton onClick={onClick}>test</LinkButton>);
    component.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });

  it('doesnt call onClick function if disabled', () => {
    const onClick = jest.fn();
    const component = shallow(<LinkButton onClick={onClick} disabled>test</LinkButton>);
    component.simulate('click');
    expect(onClick).not.toHaveBeenCalled();
  });

  it('renders child in a button tag', () => {
    const component = shallow(<LinkButton>test</LinkButton>);
    expect(component.find('button').text()).toBe('test');
  });
});
