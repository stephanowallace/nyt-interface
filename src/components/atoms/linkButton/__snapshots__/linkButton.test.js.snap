// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link button renders default state 1`] = `
<button
  className="
      linkButton
      "
  onClick={null}
  type="button"
>
  test
</button>
`;
